gitlab-pipeline/stage/build.yml

```diff
@@ -256,6 +256,9 @@ build:mender-gateway:package:
   needs:
     - init:workspace
   before_script:
+    # Early exit when building an integration version without mender-gateway
+    - tar -tf workspace.tar.gz
+        ./go/src/github.com/mendersoftware/mender-gateway  >/dev/null 2>/dev/null || exit 0
```

Comment:
I think `script` is still executed though. Is that ok?

Reply:
It is not executed (and it would not be okey, no).

GitLab concatenates both `before_script` and `script` and executes them in a single shell. A different story is with `after_script`, which is executed in a separate shell. See the docs [here](https://docs.gitlab.com/ee/ci/yaml/#before_script).

Reply:
You're right, my bad.